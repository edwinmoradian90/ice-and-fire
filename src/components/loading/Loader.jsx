import React from 'react'
import PropTypes from 'prop-types'
import {
    Text,
    Title,
    LoadingContainer,
    Container,
    center
} from '../../utils/styledComponents/main'
import RingLoader from 'react-spinners/ClipLoader'
import colors from '../../utils/colors'

const { red } = colors

export function Loader(props) {
    const { loading } = props
    return (
        <Container className="loading">
            <Title>Ice and Fire</Title>
            <LoadingContainer>
                <Text className="loadingText">Loading...</Text>
                <RingLoader
                    css={center}
                    size={60}
                    color={red}
                    loading={loading}
                />
            </LoadingContainer>
        </Container>
    )
}

Loader.propTypes = {
    loading: PropTypes.bool
}