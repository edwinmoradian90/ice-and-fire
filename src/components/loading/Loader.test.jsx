import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Loader } from './Loader';

test('Loader component functions', () => {
    const component = renderer.create(
        <Router>
            <Loader />
        </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children[0]).toContain('Ice and Fire');
    expect(tree.children[1].children[0].children[0]).toContain('Loading...');
});