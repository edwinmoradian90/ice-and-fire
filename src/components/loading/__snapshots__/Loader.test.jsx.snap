// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component functions 1`] = `
<div
  className="sc-AxirZ jRTNxu loading"
>
  <h1
    className="sc-AxhCb kYEhcE"
  >
    Ice and Fire
  </h1>
  <div
    className="sc-AxiKw iwLqeu"
  >
    <div
      className="sc-AxhUy fClwJB loadingText"
    >
      Loading...
    </div>
    <div
      className="css-19n5wh0"
    />
  </div>
</div>
`;
